go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -406,9 +406,9 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *corev1.Service, netI
 
 	// Add flows for default network services that are accessible from UDN networks
 	if util.IsNetworkSegmentationSupportEnabled() {
-		// The flow added below has a higher priority than the per UDN service flow:
-		//   priority=200, table=2, ip, ip_src=169.254.0.<UDN>, actions=set_field:<bridge-mac>->eth_dst,output:<UDN-patch-port>
-		// This ordering ensures that traffic to UDN allowed default services goes to the the default patch port.
+		// The flow added below has a higher priority than the per UDN service isolation flow:
+		//   priority=200, table=2, ip, ip_src=169.254.0.<UDN>, actions=drop
+		// This ordering ensures that traffic to UDN allowed default services goes to the default patch port.
```

Comment:
note to self: this change is OK because:
1. the original behavior implemented by @dceara  was to match on masquerade IP from the host and then dispatch the packet to the right patch port
2. This behavior was modified by @kyrtapz to set pkt marks on service packets and then use another flow to dispatch them here: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4792

So now these masq flows should not be used by anything and we should drop the traffic, rather than letting it hit the next rule which would forward to CDN patch port.